sktime/utils/_testing/deep_equals.py

```diff
@@ -87,30 +130,47 @@ def _tuple_equals(x, y):
     ----------
     x: tuple or list
     y: tuple or list
+    return_msg : bool, optional, default=False
+        whether to return informative message about what is not equal
 
     Returns
     -------
-    bool - True if x and y are equal in value
+    is_equal: bool - True if x and y are equal in value
         x and y do not need to be equal in reference
+    msg : str, only returned if return_msg = True
+        indication of what is the reason for not being equal
+            concatenation of the following elements:
+            .len - length is not equal
+            [i] - i-th element not equal
     """
+
+    def ret(is_equal, msg):
```

Comment:
This function is already defined in the previous scope.

Reply:
yes, not too DRY - although I think it´s easier to read if it´s close to the usage.